### PR TITLE
upgrade status-go and web3.js

### PR DESCRIPTION
**Pull request by rasom**
- `status-go` is upgraded to https://github.com/status-im/status-go/commit/e6e05aff01101e2ea8f4bc6765faf92706fa6faa
- `web3.js` is upgraded to 0.20.1 with our fork of `bignumber` https://github.com/status-im/bignumber.js/commit/cc066a0a3d6bfe0c436c9957f4ea8344bf963c89
- `bignumber` dependency is removed in favor of using it through `web3.js`

status: ready

### CODE REVIEW

**Comment by rasom** [comment]
This PR can be tested but can't be merged because of https://github.com/status-im/status-go/issues/251

File: src/status_im/ios/core.cljs [hunk 1]
```diff
@@ -43,7 +43,7 @@
        :reagent-render views/main})))
 
 (defn init []
-  (utils/register-exception-handler)
+  #_(utils/register-exception-handler)
```
Comment (flexsurfer) [inline src/status_im/ios/core.cljs]: ?

File: src/status_im/protocol/web3/filtering.cljs [hunk 2]
```diff
@@ -18,11 +18,14 @@
     (swap! filters update web3 dissoc options)))
 
 (defn add-shh-filter!
-  [web3 options callback]
-  (let [options' (update options :type (fn [t] (or t :asym)))
-        filter   (.filter (u/shh web3) (clj->js options')
-                          callback
-                          #(log/warn :add-filter-error options %))]
+  [web3 {:keys [key type] :as options} callback]
+  (let [type     (or type :asym)                               ;
```
Comment (flexsurfer) [inline src/status_im/protocol/web3/filtering.cljs]: ;

**Comment by rasom** [comment]
@oskarth see this commit https://github.com/status-im/web3.js/commit/c1796b6dd4539286dd9a84ddf1e23a7298b6370b
thanks @flexsurfer, will fix soon

**Comment by rasom** [comment]
should be merged only when https://github.com/status-im/status-go/issues/251 will be fixed

**Comment by yenda** [comment]
Is it still a WIP ?

**Comment by flexsurfer** [comment]
@yenda no, but it blocked to merge

**Comment by flexsurfer** [comment]
status-im/status-go#251 has been fixed, this PR can be merged after tests

**Comment by annadanchenko** [comment]
App installed from build #21 
Syncing always starts with 0%, 1% ... as reported in https://github.com/status-im/status-go/issues/185 Tried on Android and iOS in 3 different locations (Russia, Poland, Belarus) installed for at least 10 times - always syncing  from 0%.

**Comment by annadanchenko** [comment]
On latest build 
1. syncing completes in about 10 seconds on iPhone 6s
2. no new chat for contact B is shown if contact A adds B to the Contacts and sends text, request or send transaction
3. On first installation there was popup `No protocol method IAssociative. - assoc defined for type cljs.core` shown after contact A sends to B send transaction
4. After re-installing the app and repeating same steps there was no popup anymore but contact B still has no new chat created, so A and B can't communicate

tested on 2 real iOS devices: iphone 7 plus and iphone 6s
Branch: feature/upgrade-status-go-g0270590
iOS: https://i.diawi.com/W55VZD

**Comment by annadanchenko** [comment]
- last Android build for `feature/upgrade-status-go-g0270590` https://status-im.slack.com/archives/G53FWGDCM/p1504782471000086 Does not use CHT. Installed 3 times so far, on Galaxy S6  syncing started from scratch once, then reinstalled build twice there and it started from 65 % both times. On Google Pixel - 2 attempts and in both syncs from 0%.
- Same last build on iOS syncs for 10 seconds. 2 iOS devices can exchange text messages and send ETH to each other

**Comment by annadanchenko** [comment]
on iOS (not sure for Android): #1779

File: Jenkinsfile [hunk 3]
```diff
@@ -31,52 +31,57 @@ node {
...
 
   } catch (e) {
     slackSend color: 'bad', message: env.BRANCH_NAME + ' failed to build. ' + env.BUILD_URL
     throw e
   }
-
-  stage('Slack Notification') {
-    def c = (testPassed ? 'good' : 'warning' )
-    slackSend color: c, message: 'Branch: ' + env.BRANCH_NAME + '\nTests: ' + (testPassed ? ':+1:' : ':-1:') + ')\nAndroid: ' + apkUrl + '\n iOS: ' + ipaUrl
-  }
```
Comment (oskarth) [inline Jenkinsfile:81]: This `Jenkinsfile` stuff seems like part of a different PR?

**Pull request merged**